shuf: Refuse repeating zero lines

BenWiederhake opened the pull request:
This was a GNU behavior bug:

```console
$ LC_ALL=C shuf -er
shuf: no lines to repeat
[$? = 1]
$ cargo run shuf -er  # old, bad (unexpected success)
$ cargo run shuf -er  # new
shuf: no lines to repeat
[$? = 1]
```

Note that this bug already existed *before* #5980 and my other shuf-related PRs.

This should also fix the last testcase of the GNU shuf test.

sylvestre commented:
Excellent 
Could you please add a test on our side?
We don't rely on gnu's (too long)
Thanks

BenWiederhake commented:
Sure thing!

BenWiederhake commented:
Changes since last push:
- Added tests. Thanks @sylvestre!
- Note that I did NOT fix or test the other newly-discovered bug, since that's a whole new can of worms, see upcoming PR.

cakebaker commented on tests/by-util/test_shuf.rs:
```diff
@@ -624,3 +624,31 @@ fn test_shuf_multiple_input_line_count() {
 }
+
+#[test]
+fn test_shuf_repeat_empty_range() {
+    new_ucmd!()
+        .arg("-ri10-3")
+        .fails()
+        .no_stdout()
+        .stderr_only("shuf: no lines to repeat\n");
+}
```
I think this test case is incorrect. With GNU `shuf`, running `shuf -ri10-3` leads to an "invalid input range" error. I only get a "no lines to repeat" error if the difference between the first and second number is one, for example with `shuf -ri10-9`.

BenWiederhake replied:
Huh! You're right, that's weird. Apparently GNU shuf makes a distinction between empty and negative ranges I guess? Thanks for flagging this, I'll look into it some more.

BenWiederhake commented:
Changes since last push:
- Changed `test_shuf_repeat_empty_range` to test really *only* the empty range. Thanks @cakebaker!
- Set it to `#[ignore]`, as this test necessarily triggers the bug that will be resolved in #6014
- The newly-found existing "GNU behavior bug" about negative ranges will also be handled in #6014.

Btw, I don't really understand what `inotify-dir-recreate` is about, but it seems unrelated for several reasons.

cakebaker commented:
> Congrats! The gnu test tests/shuf/shuf is no longer failing!

Great, thanks!
